hudi-common/src/main/java/org/apache/hudi/common/model/OverwritePrecombineAvroPayload.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.common.model;
+
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.hudi.avro.HoodieAvroUtils;
+import org.apache.hudi.common.util.Option;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * subclass of OverwriteWithLatestAvroPayload used for delta streamer.
+ *
+ * <ol>
+ * <li>preCombine - When more than one HoodieRecord have the same HoodieKey, this function combines all fields(which is not null)
+ * before attempting to insert/upsert.
+ * eg: 1)
+ * Before:
+ * id name  age   ts
+ * 1  Karl  null  0.0
+ * 1  null  18    0.0
+ * After:
+ * id name  age   ts
+ * 1  Karl  18    0.0
+ * </ol>
+ */
+public class OverwritePrecombineAvroPayload extends OverwriteWithLatestAvroPayload {
```

Comment:
can we name this better? actually did you intend to add 2 new payloads? (given there is only test for 1)

Reply:
Because PartialAvroPayload's `precombine` is the same as OverwritePrecombineAvroPayload's. 
PartialAvroPayload's `combineAndGetUpdateValue`  is the same as OverwriteNonDefaultsWithLatestAvroPayload's.
OverwritePrecombineAvroPayload and OverwriteNonDefaultsWithLatestAvroPayload have test already.

Reply:
Because `PartialAvroPayload`'s `precombine` is the same as `OverwritePrecombineAvroPayload`'s.
`PartialAvroPayload`'s `combineAndGetUpdateValue` is the same as `OverwriteNonDefaultsWithLatestAvroPayload`'s.
`OverwritePrecombineAvroPayload` and `OverwriteNonDefaultsWithLatestAvroPayload` have test already.